node/node_handler.go

```diff
@@ -56,101 +55,75 @@ func (node *Node) processSkippedMsgTypeByteValue(
 	}
 }
 
-// HandleMessage parses the message and dispatch the actions.
-func (node *Node) HandleMessage(content []byte, sender libp2p_peer.ID) {
-	// log in-coming metrics
-	node.host.LogRecvMessage(content)
-	utils.Logger().Debug().
-		Int64("TotalIn", node.host.GetBandwidthTotals().TotalIn).
-		Float64("RateIn", node.host.GetBandwidthTotals().RateIn).
-		Msg("[metrics][p2p] traffic in in bytes")
-
-	msgCategory, err := proto.GetMessageCategory(content)
-	if err != nil {
-		utils.Logger().Error().
-			Err(err).
-			Msg("HandleMessage get message category failed")
-		return
-	}
-	msgType, err := proto.GetMessageType(content)
-	if err != nil {
-		utils.Logger().Error().
-			Err(err).
-			Msg("HandleMessage get message type failed")
-		return
-	}
-
-	msgPayload, err := proto.GetMessagePayload(content)
-	if err != nil {
-		utils.Logger().Error().
-			Err(err).
-			Msg("HandleMessage get message payload failed")
-		return
-	}
+var (
+	errInvalidPayloadSize = errors.New("invalid payload size")
+	errWrongBlockMsgSize  = errors.New("invalid block message size")
+)
 
-	switch msgCategory {
-	case proto.Consensus:
-		msgPayload, _ := proto.GetConsensusMessagePayload(content)
-		if node.NodeConfig.Role() == nodeconfig.ExplorerNode {
-			node.ExplorerMessageHandler(msgPayload)
-		} else {
-			node.ConsensusMessageHandler(msgPayload)
+// HandleNodeMessage parses the message and dispatch the actions.
+func (node *Node) HandleNodeMessage(
+	ctx context.Context,
+	payload []byte,
+) error {
+
+	// Prevent OOB crash
+	if len(payload) < (proto.MessageTypeBytes + proto.MessageTypeBytes) {
```

Comment:
Should it be MessageTypeBytes + MessageCatagoryBytes?

Reply:
fixed in new PR https://github.com/harmony-one/harmony/pull/3212